ykcapi/src/lib.rs

```diff
@@ -48,6 +52,122 @@ pub extern "C" fn __ykutil_get_llvmbc_section(res_addr: *mut *const c_void, res_
     }
 }
 
+/// Reads out registers spilled to the stack of the previous frame during the deoptimisation
+/// routine. The order of the registers are in accordance to the DWARF register number mapping
+/// referenced in the SystemV ABI manual (https://uclibc.org/docs/psABI-x86_64.pdf).
+struct Registers {
+    addr: *const usize,
+}
+
+impl Registers {
+    /// Creates a Registers struct from a given a pointer on the stack containing spilled
+    /// registers.
+    fn from_ptr(ptr: *const c_void) -> Registers {
+        Registers {
+            addr: ptr as *const usize,
+        }
+    }
+
+    /// Read the spilled register value at the offset `off` from the previous stack frame.
+    fn read_from_stack(&self, off: isize) -> usize {
+        unsafe { ptr::read::<usize>(self.addr.offset(off)) }
```

Comment:
Is it intended that the offset uses pointer arithmetic instead of integer arithmetic?

Reply:
Yes, `addr` is a pointer into the stack where we read from different offsets. Why is it important to make clear that we are using pointer arithmetic here?

Reply:
I initially got confused. If you don't want to change the docs that is fine too.